### PR TITLE
dotnet-sdk: 6.0.412 -> 6.0.413, dotnet-sdk_7: 7.0.306 -> 7.0.400

### PR DESCRIPTION
**Pull request by mdarocha**
## Description of changes

https://github.com/dotnet/core/releases/tag/v7.0.10
https://github.com/dotnet/core/releases/tag/v6.0.21

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

@NixOS/dotnet

### CODE REVIEW

**Review by JamieMagee — APPROVED** [review]

**Review by corngood — APPROVED** [review]

**Comment by AlexeyRaga** [comment]
Can this be merged?

**Comment by wegank** [comment]
@ofborg build dotnet-sdk dotnet-sdk.passthru.tests dotnet-sdk_7 dotnet-sdk_7.passthru.tests

**Comment by wegank** [comment]
@ofborg build python310Packages.clr-loader

File: pkgs/development/compilers/dotnet/versions/6.0.nix [hunk 1]
```diff
@@ -3,177 +3,177 @@
...
+      (fetchNuGet { pname = "Microsoft.NETCore.App.Runtime.linux-musl-arm64"; version = "6.0.21"; sha256 = "1cwq84i6xd4h2ddl4yjzd6pbf3cm3g2kids0xkgjcafx3qjiqg4b"; })
+      (fetchNuGet { pname = "Microsoft.NETCore.App.Runtime.linux-musl-x64"; version = "6.0.21"; sha256 = "0pgc2rzsr1ag8hk1g75w1sm3p0qabr7fddckfsfk4sk54gwdr14k"; })
+      (fetchNuGet { pname = "Microsoft.NETCore.App.Runtime.linux-x64"; version = "6.0.21"; sha256 = "13f65051h35hbvrz2g26vsvi9c1fbrc1hkjsxvx2ys7s1qddddqw"; })
+      (fetchNuGet { pname = "Microsoft.NETCore.App.Runtime.osx-x64"; version = "6.0.21"; sha256 = "0y9dwb9fpycff0jh3c7gqmz9vrbg52p1bs4d437b73sfl7yj6s0x"; })
+      (fetchNuGet { pname = "Microsoft.NETCore.App.Runtime.win-arm"; version = "6.0.21"; sha256 = "1hww6k853nk8i620d9gmldbaj9ylawwqhzcjfmzpkmpj1hij80jd"; })
+      (fetchNuGet { pname = "Microsoft.NETCore.App.Runtime.win-arm64"; version = "6.0.21"; sha256 = "0ihsszn9s3dsk31rr4qvh3kvlf3dwrzcb0s91dblcgi40gyczqb3"; })
+      (fetchNuGet { pname = "Microsoft.NETCore.App.Runtime.win-x64"; version = "6.0.21"; sha256 = "0qpzjnmhc5qp6y53jmrv4b675nafw5yrz5b45ay91rgvf26x82pv"; })
```
Comment (wegank) [inline pkgs/development/compilers/dotnet/versions/6.0.nix:101]: Hash mismatch 😕

https://logs.ofborg.org/?key=nixos/nixpkgs.248208&attempt_id=67b90b0c-bef7-4ece-9c5f-512fed77a8ef

Comment (mdarocha) [reply pkgs/development/compilers/dotnet/versions/6.0.nix:101]: fixed

**Review by wegank — APPROVED** [review]

**Comment by wegank** [comment]
```
1 package failed to build:
tests.writers.simple.fsharp

86 packages built:
ArchiSteamFarm BeatSaberModManager alttpr-opentracker audiobookshelf avalonia-ilspy azure-functions-core-tools bililiverecorder boogie btcpayserver btcpayserver-altcoins certdump csharp-ls dafny denaro depotdownloader discordchatexporter-cli dotnet-aspnetcore dotnet-aspnetcore_7 dotnet-runtime dotnet-runtime_7 dotnet-sdk dotnet-sdk_7 eventstore fable famistudio formula fsautocomplete galaxy-buds-client git-credential-manager github-runner godot3-mono godot3-mono-debug-server godot3-mono-export-templates godot3-mono-headless godot3-mono-headless.dev godot3-mono-headless.man godot3-mono-server godot3-mono.dev godot3-mono.man ilspycmd inklecate jackett jellyfin jetbrains.rider kavita lidarr marksman mqttmultimeter msbuild nbxplorer netcoredbg networkminer nuget-to-nix omnisharp-roslyn openra opentabletdriver openutau osu-lazer pbm pinta pocket-updater-utility pre-commit pre-commit.dist prowlarr ps3-disc-dumper python310Packages.clr-loader python310Packages.clr-loader.dist python310Packages.pythonnet python310Packages.pythonnet.dist python311Packages.clr-loader python311Packages.clr-loader.dist python311Packages.pythonnet python311Packages.pythonnet.dist radarr readarr roslyn ryujinx scarab slskd space-station-14-launcher tests.dotnet.project-references tests.writers.simple.fsharpNoNugetDeps tone vintagestory wasabibackend xivlauncher
```

**Pull request merged**